src/main/webapp/app/entities/submission.model.ts

```diff
@@ -62,36 +68,26 @@ export function getLatestSubmissionResult(submission: Submission | undefined): R
  * @param submission
  * @param result
  *
- * @return the index of the latestSubmissionResult or undefined if any of the params was undefined
+ * @return
  */
-export function setLatestSubmissionResult(submission: Submission | undefined, result: Result | undefined): number | undefined {
-    checkForResultsLength(submission);
-
+export function setLatestSubmissionResult(submission: Submission | undefined, result: Result | undefined) {
     if (!submission || !result) {
-        return undefined;
+        return;
     }
 
     if (submission!.results && submission!.results.length > 0) {
         submission.results[submission.results.length - 1] = result;
     } else {
         submission.results = [result];
     }
-    return submission.results.length - 1;
+    submission.latestResult = result;
```

Comment:
Quick Note, the method setLatestSubmissionResult isn't actually used in the code right now, at the moment we just do `submission.latestResult = getlatestSubmissionResult(submission);`
-> I think we have two option: remove the function, or rewrite the calls of  
`submission.latestResult = getlatestSubmissionResult(submission);` to 
`setLatestSubmissionResult(submission, getLatestSubmissionResult(submission))`